BE sending order

lei29 opened the pull request:
order model, get order from both chef or customers, post order and chef and customer store the order, put request to change the status of order, like open, accepted, paid, closed... **Currently only got 2 types , open, closed.**

Commit:
order model added

Commit:
API for orders

get orders for either chef or customer, post order and store it in chef and customer.

Commit:
put request to set status

Commit:
Merge branch 'dev' into BE-sending-order

smithpg commented on server/routes/order.js:
```diff
@@ -0,0 +1,61 @@
+const Joi = require("joi");
+const createError = require("http-errors");
+const { decodeToken, userIsAuthorized } = require("../middleware/auth");
+const fileUploadService = require("../services/fileUploader");
```
```suggestion

```

smithpg commented on server/routes/order.js:
```diff
@@ -0,0 +1,61 @@
+    if (error) return next(createError(400, error.details[0].message));
+
+    const order = await Order.create(req.body);
+    const chef = await Chef.findById(req.params.chef_id);
```
As an alternative, we could add a pre('save') hook to Order model similar to the hook that exists on the Dish model for adding new dish to its chef's document

smithpg commented on server/routes/order.js:
```diff
@@ -0,0 +1,61 @@
+
+    const user = await User.findById(req.params.user_id);
+
+  res.status(200).send(user.orders);
```
Does this send the actual order objects, or just their id's? I think the default behavior when you retrieve a document that has ObjectIds stored on it is that you need to call populate on the returned model in order to get the other documents referenced by the ids.

lei29 replied:
Got it i will change that

Commit:
Return order object

changed typo and enable createdat and updatedat property and use populate to return real object
